File: Miscelaneous Questions/Algorithms in C/isLeapYear.c
#include <stdio.h>

int main()
{
	int year;
	
	printf("Please enter the year: ");
	scanf("%d", &year);
	
	if (year % 4 == 0)
	{
		if (year % 100 == 0)
		{
			if (year % 400 == 0)
				printf("%d is a leap year\n", year);
			else
				printf("%d is not a leap year\n", year);
		}
		else
			printf("%d is a leap year\n", year);
	}
	else
		printf("%d is not a leap year\n", year);
	
	return 0;
}

File: Miscellaneous Questions/Algorithms in Python/isLeapYear.py
# Program to determine if a given year is a leap year

# Request user input
year = int(input("Enter a year: "))

# Check if year is divisible by 4
if (year % 4) == 0:
   if (year % 100) == 0:
       if (year % 400) == 0:
           print("{0} is a leap year".format(year))
       else:
           print("{0} is not a leap year".format(year))
   else:
       print("{0} is a leap year".format(year))
else:
   print("{0} is not a leap year".format(year))